src/main/java/by/exadel/internship/repository/location/CityRepository.java

```diff
@@ -2,9 +2,17 @@
 
 import by.exadel.internship.entity.location.City;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 @Repository
 public interface CityRepository extends JpaRepository<City, UUID> {
+
+    @Query("SELECT DISTINCT c FROM City c LEFT JOIN FETCH c.country WHERE c.country.name = :countryName")
```

Comment:
I think you can do this without @Query. Check this link https://docs.spring.io/spring-data/jpa/docs/1.5.0.RELEASE/reference/html/jpa.repositories.html

Reply:
I think no, because I if i do findAllByCountry(Country country) i need to pass Country, but I wanted to pass String name

Reply:
@OlgaEvdokimova You can solve this restriction using this way -> https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-property-expressions

Reply:
thanks, done

Reply:
@OlgaEvdokimova @Query annotation still in place

Reply:
I dropped it